client-src/src/pages/counselor/follow-up-modal/follow-up-modal.ts

```diff
@@ -0,0 +1,142 @@
+import { Component } from '@angular/core';
+import { NavController, NavParams, ViewController } from 'ionic-angular';
+import { FormGroup } from '@angular/forms';
+import { FormBuilder } from '@angular/forms';
+import { Validators } from '@angular/forms';
+import { DatePipe } from '@angular/common';
+
+import { FollowUpProvider } from '../../../providers/follow-up/follow-up'
+import { NotificationProvider } from '../../../providers/notification/notification';
+import { NotificationMessageProvider } from '../../../providers/notification-message/notification-message';
+import { HelperProvider } from '../../../providers/helper/helper';
+
+@Component({
+  selector: 'page-follow-up-modal',
+  templateUrl: 'follow-up-modal.html',
+})
+export class FollowUpModalPage {
+
+  private enums;
+  private followUpStatus;
+  private subStatus;
+  private followUpType;
+  private caseIndex;
+  private inquiryId;
+  private inquiryName;
+  private currentFollowUp;
+
+  private followUpForm: FormGroup
+
+  constructor(public navCtrl: NavController, public navParams: NavParams, private view: ViewController,private formBuilder: FormBuilder,private followUpProvider: FollowUpProvider, private notify: NotificationProvider, private message: NotificationMessageProvider, private datePipe: DatePipe, private helper: HelperProvider) {
+
+    this.followUpForm = this.formBuilder.group({
+      followUpType: ['', Validators.required],
+      followUpStatus: ['', Validators.required],
+      remark: [''],
+      caseIndex: ['', Validators.required],
+      subStatus: ['', Validators.required]
+    });
+
+    this.inquiryId = this.navParams.get('id');
+    this.inquiryName = this.navParams.get('name');
+
+    this.patchForm();
+    this.setEnums();
+    this.setFollowUpStatus();
+    this.setSubStatus();
+    this.setFollowUpType();
+    this.setCaseIndex();
+
+  }
+
+  ionViewDidLoad() {
+    console.log('ionViewDidLoad FollowUpModalPage');
+  }
+
+  createFollowUp(){
```

Comment:
Merge create and update functions here and call the merged function from html. Inside the merged function check if id is available then based on that request update API